<launch>
  <arg name="debug" default="false"/>
  <arg name="globle_path" default="/elevator_path"/>
  <arg name="costmap_topic" default="/campusrover_local_costmap"/>
  <arg name="twist_output_topic" default="/yocs_cmd_vel_mux/input/navigation"/>
  <arg name="enable_path_switch_stop" default="false"/>
  <arg name="enble_costmap_obstacle" default="true"/>
  <arg name="avoid_right_side_priority" default="true"/>
  

 <!-- /////////////////////////////////////////////////////////////////////////////////// -->
 <group if="$(arg debug)">

   <!-- <param name="use_sim_time" value="true" /> -->

   <param name="robot_description"
     textfile="$(find campusrover_description)/urdf/campusrover.urdf" />

   <node name="robot_state_publisher" pkg="robot_state_publisher"
     type="robot_state_publisher" />


   <!-- rviz -->
   <node name="rviz" pkg="rviz" type="rviz" args="-d $(find campusrover_move)/rviz/dp_planner.rviz" required="true"/>

 </group>

  <!-- dp planner -->
  <node pkg="campusrover_move" type="adp_planner" name="adp_planner" output="screen">
    <remap from="path"    to="$(arg globle_path)" />
    <remap from="costmap" to="$(arg costmap_topic)" />
    <remap from="cmd_vel" to="$(arg twist_output_topic)" />
    <param name="robot_frame" value="base_link" />
    <param name="arriving_range" value="0.2" />
    <param name="max_vel_x" value="0.4" />
    <param name="max_rot_vel" value="3.0" />
    <param name="min_rot_vel" value="-3.0" />
    <param name="min_turning_radius" value="0.0" />

    <param name="step_linear_x" value="0.05" />
    <param name="step_angular_z" value="0.02" /> 
    <param name="angular_velocity_p" value="1.2" /> 
  
    <param name="max_dist" value="0.3" />
    <param name="min_dist" value="0.3" />
    <param name="vel_x_step" value="0.8" />
    <param name="rot_x_step" value="0.01" /> 
    <param name="rot_x_zero_vel_angle" value="3.0" /> 
    <param name="path_resolution" value="0.1" />
    <param name="swap_path_num" value="20" />

    <param name="target_point_dis" value="1.2" />
    <param name="threshold_occupied" value="10" />

    <param name="footprint_max_x" value="0.4" />
    <param name="footprint_min_x" value="-0.1" />
    <param name="footprint_max_y" value="0.01" />
    <param name="footprint_min_y" value="-0.03" />

    <param name="avoid_right_side_priority" value="$(arg avoid_right_side_priority)" />
    <param name="enable_path_switch_stop" value="$(arg enable_path_switch_stop)" />
    <param name="obstacle_path_switch_stop_time" value="1.5" />

    <param name="enble_costmap_obstacle" value="$(arg enble_costmap_obstacle)" />
    <param name="twist_profile" value="true" />
    <param name="PathVisualize" value="true" />

  </node>

</launch>
